src/utils/aiScoring.ts

```diff
@@ -0,0 +1,71 @@
+// src/utils/aiScoring.ts
+import type { ProgressBarData } from "@/utils/models";
+import { ID_TO_STYLE_REMAP, STYLE_TO_CATEGORY_MAP } from "./aiConstants";
+
+export const calculateScoresAndProgressBarData = (
+  probs: number[],
+  movementScore: number,
+  rhythmScore: number,
+): ProgressBarData[] => {
+  let slowSum = 0, fastSum = 0;
+  for (let i = 0; i < probs.length; i++) {
+    const style = ID_TO_STYLE_REMAP[i];
+    const categories = STYLE_TO_CATEGORY_MAP[style] || [];
+    const prob = probs[i];
+    if (categories.includes("slow")) slowSum += prob;
+    if (categories.includes("fast")) fastSum += prob;
+  }
+  
+  const aiTempoTotal = slowSum + fastSum;
+  const aiTempoScore = aiTempoTotal > 0 ? (fastSum / aiTempoTotal) * 100 : 50;
+
+  const AI_TEMPO_WEIGHT = 0.3;
+  const DYNAMICS_TEMPO_WEIGHT = 0.5;
+  const combinedTempoScore = (aiTempoScore * AI_TEMPO_WEIGHT) + (rhythmScore * DYNAMICS_TEMPO_WEIGHT);
+
+  const expressionScore = (combinedTempoScore + movementScore) / 2;
+  
+  const applyContrast = (value: number): number => {
+    const normalized = value / 100;
+    const adjusted = 0.5 + 1.5 * (normalized - 0.5);
+    const result = Math.max(0, Math.min(1, adjusted)) * 100;
+    return 50 + (result - 50);
+  };
+
+  const bars: ProgressBarData[] = [
+    {
+      labelLeft: "流麗な",
+      labelRight: "リズミカルな",
+      value: applyContrast(combinedTempoScore),
+    },
+    { 
+      labelLeft: "小さな動き",
+      labelRight: "大きな動き", 
+      value: applyContrast(movementScore)
+    },
+    { 
+      labelLeft: "単調な表現",
+      labelRight: "豊かな表現", 
+      value: applyContrast(expressionScore) 
```

Comment:
`単調な`だと悪い意味になってしまいそうなので、このように表現したらどうでしょうか？
`小さな動き`もあまりいい表現ではない気がするので、このように変更したいです！
```suggestion
    { 
      labelLeft: "密やかな動き",
      labelRight: "壮大な動き", 
      value: applyContrast(movementScore)
    },
    { 
      labelLeft: "安定した表現",
      labelRight: "豊かな表現", 
      value: applyContrast(expressionScore) 
```